storage/innobase/fts/fts0fts.cc

```diff
@@ -2607,30 +2607,31 @@ fts_get_next_doc_id(
 	return(DB_SUCCESS);
 }
 
-/*********************************************************************//**
-This function fetch the Doc ID from CONFIG table, and compare with
+/** This function fetch the Doc ID from CONFIG table, and compare with
 the Doc ID supplied. And store the larger one to the CONFIG table.
+@param table fts  table
+@param cmp_doc_id Doc ID to compare
+@param read_only  TRUE if read the synced_doc_id only
+@param doc_id     larger document id after comparing "cmp_doc_id" to
+	          the one stored in CONFIG table
+@param trx	  transaction
 @return DB_SUCCESS if OK */
-static MY_ATTRIBUTE((nonnull))
+static
 dberr_t
 fts_cmp_set_sync_doc_id(
-/*====================*/
-	const dict_table_t*	table,		/*!< in: table */
-	doc_id_t		cmp_doc_id,	/*!< in: Doc ID to compare */
-	ibool			read_only,	/*!< in: TRUE if read the
-						synced_doc_id only */
-	doc_id_t*		doc_id)		/*!< out: larger document id
-						after comparing "cmp_doc_id"
-						to the one stored in CONFIG
-						table */
+	const dict_table_t *table,
+	doc_id_t           cmp_doc_id,
+	ibool	           read_only,
+	doc_id_t           *doc_id,
+	trx_t	           *trx=nullptr)
 {
-	trx_t*		trx;
 	pars_info_t*	info;
 	dberr_t		error;
 	fts_table_t	fts_table;
 	que_t*		graph = NULL;
 	fts_cache_t*	cache = table->fts->cache;
 	char		table_name[MAX_FULL_NAME_LEN];
+	bool		created_trx = (trx == nullptr);
```

Comment:
We can save one operation:
```c++
	const trx_t* const created_trx = trx;
```
and adjust each check for `!created_trx` to `created_trx`.